input/pagecontent/smart-web-messaging.md

```diff
@@ -388,15 +410,255 @@ targetWindow.postMessage({
+scratchpad.
+
+```js
+// From app -> EHR
+ehrWindow.postMessage({
+  "messageId": "<some new uid>",
+  "messagingHandle": "<smart_web_messaging_handle> from SMART launch context",
+  "messageType": "scratchpad.read"
+}, ehrOrigin);
+```
+
+The EHR could then respond with the full contents of the scratchpad in one
+bundle.
+
+```js
+// From EHR -> app
+appWindow.postMessage({
+  "messageId": "<some new uid>",
+  "responseToMessageId": "<corresponding request messageId>",
+  "payload": {
+    "status": "200 OK",
+    "resources": {
+      "id": "<some new uid>",
+      "resourceType": "Bundle",
+      "type": "collection",
+      "entry": [
+        {
+          "fullUrl": "ServiceRequest/1",
+          "resource": {
+            "resourceType": "ServiceRequest",
+            "id": "1",
+            "status": "draft",
+            "intent": "proposal",
+            "subject": {
+              "reference": "http://example.com/Patient/123"
+            }
+          }
+        },
+        {
+          "fullUrl": "MedicationRequest/1",
+          "resource": {
+            "resourceType": "MedicationRequest",
+            "id": "1",
+            "status": "draft",
+            "intent": "proposal",
+            "medicationCodeableConcept": {
+              "coding": [
+                {
+                  "system": "http://snomed.info/sct",
+                  "code": "108761006",
+                  "display": "Capecitabine-containing product"
+                }
+              ]
+            },
+            "subject": {
+              "reference": "http://example.com/Patient/123"
+            }
+          }
+        }
+      ]
+    }
+  }
+}, appOrigin);
 ```
 
+---
+
+This example shows how an app might inspect the entire contents of the
+scratchpad.
+
+```js
+// From app -> EHR
+ehrWindow.postMessage({
+  "messageId": "<some new uid>",
+  "messagingHandle": "<smart_web_messaging_handle> from SMART launch context",
+  "messageType": "scratchpad.read"
+}, ehrOrigin);
+```
+
+Assuming the scratchpad is empty, the EHR could respond to the app with:
+
+```js
+// From EHR -> app
+appWindow.postMessage({
+  "messageId": "<some new uid>",
+  "responseToMessageId": "<corresponding request messageId>",
+  "payload": {
+    "status": "200 OK"
+  }
+}, appOrigin);
+```
+
+---
+
+This example shows how an app might request the contents of a single resource
+from the scratchpad, `ServiceRequest/321`.
+
+```js
+// From app -> EHR
+ehrWindow.postMessage({
+  "messageId": "<some new uid>",
+  "messagingHandle": "<smart_web_messaging_handle> from SMART launch context",
+  "messageType": "scratchpad.read",
+  "payload": {
+    "location": "ServiceRequest/321"
+  }
+}, ehrOrigin);
+```
+
+Assuming the resource does *not* exist, the EHR could return the following:
+
+```js
+// From EHR -> app
+appWindow.postMessage({
+  "messageId": "<some new uid>",
+  "responseToMessageId": "<corresponding request messageId>",
+  "payload": {
+    "status": "404 NOT FOUND"
```

Comment:
If we're thinking about this scratchpad.read like a FHIR search, then `404` seems wrong -- it'd be like saying "Sorry, this server doesn't have a search endpoint". We want to say "you've successfully read zero resources", right? In which case, we'd populate an empty `response` (or a `response` with zero `entry`s -- see my other comment on that point).

Reply:
This got me thinking that the `status` attribute is not really adding much value for the `read` operations.  I've removed it and updated the examples.